src/confluent_kafka/aio/_AIOProducer.py

```diff
@@ -60,47 +142,254 @@ async def _call(self, blocking_task, *args, **kwargs):
+            }
+        
+        # Calculate average latency
+        avg_latency = self._total_latency / self._total_calls
+        
+        # Calculate median latency
+        sorted_latencies = sorted(self._latency_metrics)
+        n = len(sorted_latencies)
+        if n % 2 == 0:
+            median_latency = (sorted_latencies[n//2 - 1] + sorted_latencies[n//2]) / 2
+        else:
+            median_latency = sorted_latencies[n//2]
+        
+        # Calculate percentiles
+        p95_index = int(0.95 * n)
+        p95_latency = sorted_latencies[min(p95_index, n-1)]
+        
+        p99_index = int(0.99 * n)
+        p99_latency = sorted_latencies[min(p99_index, n-1)]
+        
+        return {
+            'total_calls': self._total_calls,
+            'total_latency': self._total_latency,
+            'avg_latency': avg_latency,
+            'median_latency': median_latency,
+            'min_latency': self._min_latency if self._min_latency != float('inf') else 0.0,
+            'max_latency': self._max_latency,
+            'p95_latency': p95_latency,
+            'p99_latency': p99_latency
+        }
+
+      
+    # ========================================================================
+    # BATCHING OPERATIONS - Buffer management and batch processing
+    # ========================================================================
+
+    def _produce_batch_and_poll(self, target_topic, batch_messages, batch_delivery_callback):
+        """Helper method to run produce_batch and poll in the same thread pool worker
+        
+        This optimization combines both operations in a single thread pool execution:
+        1. produce_batch() - queues messages in librdkafka
+        2. poll(0) - immediately processes delivery callbacks
+        
+        Benefits:
+        - Reduces thread pool overhead (1 call instead of 2)
+        - Eliminates context switching between operations
+        - Ensures immediate callback processing
+        - Lower latency for delivery confirmation
+        """
+        # Call produce_batch first
+        self._producer.produce_batch(target_topic, batch_messages, on_delivery=batch_delivery_callback)
+        
+        # Immediately poll to process delivery callbacks in the same worker
+        poll_result = self._producer.poll(0)
+        
+        return poll_result
+
+    async def _flush_buffer(self, target_topic=None):
+        """Flush the current message buffer using produce_batch via thread pool
+        
+        This method now properly handles messages for different topics by grouping
+        them and calling produce_batch separately for each topic.
+        """
+        if not self._message_buffer:
+            return
+        
+        # Group messages by topic
+        topic_groups = {}
+        for i, msg_data in enumerate(self._message_buffer):
+            topic = msg_data['topic']
+            if topic not in topic_groups:
+                topic_groups[topic] = {
```

Comment:
Uhg, we're reliant on index ordering because batch produce librdkafka argument pattern aren't we